blockchain/chain/src/store/chain_store.rs

```diff
@@ -192,6 +187,34 @@ where
         }
         Ok(())
     }
+
+    pub async fn sub_head_changes(&mut self) -> Result<Receiver<HeadChange>, Error> {
+        let subscribed_head_change = self.subscribe();
+        info!("sub head changes");
+        let head = self
+            .heaviest_tipset()
+            .ok_or_else(|| Error::Other("Could not get heaviest tipset".to_string()))?;
+        let (mut sender, reciever) = channel(16);
+        sender
+            .send(HeadChange::Current(head))
+            .await
+            .map_err(|e| Error::Other(format!("Could not send to channel {:?}", e)))?;
```

Comment:
This just errors when all receivers have been dropped, are you sure you want to error here?

Reply:
No harm in leaving it aside from if there is any other problem into sending to a channel